snapshot/snapshot_test.go

```diff
@@ -0,0 +1,145 @@
+package snapshot
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/PlakarKorp/plakar/appcontext"
+	"github.com/PlakarKorp/plakar/caching"
+	"github.com/PlakarKorp/plakar/encryption/keypair"
+	"github.com/PlakarKorp/plakar/logging"
+	"github.com/PlakarKorp/plakar/repository"
+	"github.com/PlakarKorp/plakar/snapshot/importer/fs"
+	"github.com/PlakarKorp/plakar/storage"
+	bfs "github.com/PlakarKorp/plakar/storage/backends/fs"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+func randFileName(prefix string) string {
+	const chars = "abcdefghijklmnopqrstuvwxyz0123456789"
+	suffix := make([]byte, 8)
+	for i := range suffix {
+		suffix[i] = chars[rand.Intn(len(chars))]
+	}
+	return prefix + string(suffix)
+}
+
+func generateSnapshot(t *testing.T, keyPair *keypair.KeyPair) *Snapshot {
+	// init temporary directories
+	tmpRepoDir := fmt.Sprintf("/tmp/%s", randFileName("tmp_repo_"))
```

Comment:
can you please use a temporary directory like you're doing in the other cases?

For tests it should be good enough, but i prefer not to have bad examples in the code, and `randFileName` is not the concurrency-safe way of creating a temporary file. (multiple processes could still end up with the same name)

Reply:
you could just create a temporary directory and create the repository at a well-known path inside the temporary directory; in shell pseudo-code

```
tmp=$(mktemp -d)
plakar create "$tmp/repo"
```

Reply:
fixed in https://github.com/PlakarKorp/plakar/pull/345/commits/3b11f6531c36d3de118ab8199b38a9840ed4d4bc